### PR TITLE
Port optional interoperability section from MP JWT Auth spec

### PR DESCRIPTION
**Pull request by ayoho**
Adds the ["Recommendations for Optional Container Integration" section from the MP JWT Auth spec](https://github.com/eclipse/microprofile-jwt-auth/blob/main/spec/src/main/asciidoc/interoperability.asciidoc#recommendations-for-optional-container-integration) to this specification, without the "optional" qualifier.

For #4

### CODE REVIEW

**Comment by ayoho** [comment]
@sberyozkin Let me know if you think this should be in a separate interoperability.asciidoc file just like MP JWT Auth.

**Comment by sberyozkin** [comment]
Thanks @ayoho

**Review by sberyozkin — COMMENTED** [review]
@ayoho Do the tests which you have already moved support this text ? Please check, we should create issues to have whatever part of that text that is not already covered,  not immediately with this PR, but to have some plan.

Thanks for adding it to the existing doc, we can have it moved to a standalone one later if needed,  cheers

File: spec/src/main/asciidoc/microprofile-jwt-bridge-spec.asciidoc [hunk 1]
```diff
@@ -17,3 +17,43 @@
...
+
+The requirements of how a JWT should be exposed via the various Jakarta EE container APIs is discussed in this section.
+
+### Recommendations for Container Integration
+This section describes the expected behaviors for Jakarta EE container APIs other than JAX-RS.
+
+#### jakarta.security.enterprise.identitystore.IdentityStore.getCallerGroups(CredentialValidationResult)
+This method should return the set of names found in the `"groups"` claim in the JWT if it exists, an empty set otherwise.
+
+#### jakarta.ejb.SessionContext.getCallerPrincipal()
```
Comment (sberyozkin) [inline spec/src/main/asciidoc/microprofile-jwt-bridge-spec.asciidoc]: @ayoho Adam, this text got me thinking, will it make it difficult for example, for `smallrye-jwt` to offer a simple MP JWT Bridge support to be activated with #13. I wonder if this text should remain optional for now. I'll need to investigate on our end

**Comment by sberyozkin** [comment]
Hey @ayoho After looking at this PR again, it feels like it is not about the bridge as such, therefore keeping in a standalone doc, and for the moment as optional (which can revisit before the release) would work well IMHO, give a try please and refactor this PR a bit

**Comment by sberyozkin** [comment]
@ayoho As we have agreed, I've moved the optional spec text to its own doc, and also removed `configuration.asciidoc` - I believe the MP JWT Bridge spec will state that the annotation is used and then each annotation property will be described

**Review by sberyozkin — APPROVED** [review]

**Pull request merged**